### PR TITLE
passing port argument from CLI to server  #1494

### PR DESCRIPTION
**Pull request by 7flash**

### CODE REVIEW

**Comment by 7flash** [comment]
The following changes has been made..

First.. port argument is accepted in start command.. wildcard is used to ignore the actual value..

image

Second.. all arguments are passed from CLI to the server command.. 

image

**Comment by 7flash** [comment]
Consider #1 to pass other args.. Consider #2 to do the same for Dev command..

**Comment by 7flash** [comment]
Testing:

```
cd my-backend
RUST_LOG=debug ../target/debug/chisel start --internal-routes-listen-addr=127.0.0.1:9095
```

image

**Comment by 7flash** [comment]
https://github.com/chiselstrike/chiselstrike/issues/1481 https://github.com/chiselstrike/chiselstrike/issues/1494

**Comment by 7flash** [comment]
merge can be blocked by https://github.com/chiselstrike/chiselstrike/issues/1496

**Comment by dekimir** [comment]
Thank you for this, @7flash.  May I ask that you provide a description in the commit message rather than the PR comments.  You should also put "Fixes #1494" in there, so GitHub connects the two.

An easy way to edit your commit message is `git commit --amend` from the command line in your `cli-args` branch.

File: cli/src/main.rs [hunk 1]
```diff
@@ -78,7 +79,10 @@ enum Command {
         auto_index: bool,
     },
     /// Start the ChiselStrike server.
-    Start,
+    Start {
+        #[structopt(short, long, default_value="127.0.0.1:9090")]
+        internal_routes_listen_addr: SocketAddr
```
Comment (dekimir) [inline cli/src/main.rs:84]: Why is this added?  It doesn't seem to be used anywhere.

**Review by dekimir — COMMENTED** [review]
I wonder what would be the best way to test this?  Can be another PR later.

File: cli/src/main.rs (hunk 1 above)
Comment (7flash) [reply cli/src/main.rs:84]: otherwise getting an error "error[E0027]: pattern does not mention field `internal_routes_listen_addr`"..

Comment (7flash) [reply cli/src/main.rs:84]: note that currently it would give similar error when trying to pass "chisel --rpc_listen_addr".. all supported arguments should be defined here..

Comment (dekimir) [reply cli/src/main.rs:84]: Ah, OK.  Then I suggest using [external_subcommand](https://docs.rs/structopt/latest/structopt/index.html#external-subcommands) here, to eliminate such errors.

**Comment by 7flash** [comment]
@dekimir regarding your suggestion.. external_subcommand throws "enums-only" error..

image

image

**Comment by 7flash** [comment]
Nevermind.. it works when applied properly..

File: cli/src/server.rs [hunk 2]
```diff
@@ -26,15 +26,17 @@ pub(crate) fn start_server() -> anyhow::Result<std::process::Child> {
...
     cmd.pop();
     cmd.push("chiseld");
-    let server = match std::process::Command::new(cmd.clone()).spawn() {
-        Ok(server) => server,
-        Err(e) => {
-            match e.kind() {
-                ErrorKind::NotFound => anyhow::bail!("Unable to start the server because `chiseld` program is missing. Please make sure `chiseld` is installed in {}", cmd.display()),
-                _ => anyhow::bail!("Unable to start `chiseld` program: {}", e),
+    let server = match std::process::Command::new(cmd.clone())
+        .args(std::env::args().skip(2))
```
Comment (dekimir) [inline cli/src/server.rs:30]: Nitpick: this is fragile.  It will break if `start` doesn't immediately follow `chisel`, which is allowed (though currently unlikely to happen -- but we don't know what the future brings).

**Review by dekimir — APPROVED** [review]
Looks good.  The one nitpick I have can be addressed in a follow-up.

It also needs a test, which we can work on later.
